onnxruntime/contrib_ops/cuda/transformers/generation_cuda_impl.cu

```diff
@@ -1315,6 +1315,81 @@ template void BufferExpansionKernelLauncher(const int32_t* input,
                                             int chunk_size,
                                             cudaStream_t stream);
 
+constexpr unsigned int kTileSize = 32;
+constexpr unsigned int kSeqTileSize = 16;
+
+template <typename T>
+__global__ void ReorderPastStatesKernel(T* out_buffer,
+                                        const T* in_buffer,
+                                        int batch_size,
+                                        int num_heads,
+                                        int max_length,
+                                        int chunked_head_size,
+                                        int equv_chunk_size) {
+  //[B, N, max_length, H2(head_size/chunk_size), chunk_size] -> [B, N, H2(head_size/chunk_size), max_length, chunk_size]
+  __shared__ T tile[kSeqTileSize][kTileSize + 1];
```

Comment:
why `kTileSize + 1` instead of kTileSize?

Reply:
it is to avoid bank conflicts.